terraform/shared/opensearch/opensearch.tf

```diff
@@ -0,0 +1,111 @@
+resource "aws_ssm_parameter" "opensearch" {
+  name  = var.service_name
+  type  = "String"
+  value = local.connection_config
+
+  tags = merge(
+    var.tags,
+    {
+      Name = "parameter-storage"
+    }
+  )
+}
+
+resource "random_password" "es_password" {
+  length      = 16
+  special     = true
+  numeric     = true
+  upper       = true
+  lower       = true
+  min_lower   = 1
+  min_numeric = 1
+  min_special = 1
+  min_upper   = 1
+}
+
+resource "aws_security_group" "es" {
```

Comment:
As in `resource "aws_ssm_parameter" "opensearch"` let's use here also "opensearch" instead of es